resources/helm/dask-gateway/templates/gateway/deployment.yaml

```
@@ -54,6 +54,10 @@ spec:
             - mountPath: /etc/dask-gateway/
               name: configmap
           env:
+            {{- range .Values.gateway.env }}
+            - name: {{ .name }}
+              value: {{ .value| quote }}
+            {{- end }}
```

Like this, a user can provide any kind of `env` as examplified by for example https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/ or via https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#envvar-v1-core which is the definitive description of what should be allowed.

```suggestion
            {{- with .Values.gateway.env }}
            {{- . | toYaml | nindent 12 }}
            {{- end }}
```